### PR TITLE
Add unittests specifically for policy & DB interactions

### PR DESCRIPTION
**Pull request by jgadling**
### Summary:
- **What:** Add a layer of tests specifically for policy application to DB queries
- **Ticket:** [sc<fill_in_issue_number>](https://app.shortcut.com/genepi/story/<fill_in_issue_number>)
- **Env:** `<rdev link>`

### Checklist:
- [x] I merged latest `<base branch>`
- [x] I manually verified the change
- [x] I added labels to my PR
- [ ] I tested in multiple browsers
- [x] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/backend/aspen/api/policy.polar [hunk 1]
```diff
@@ -9,22 +9,19 @@ resource Group {
 
 resource Sample {
   roles = ["admin", "viewer", "member"];
-  permissions = [ "read", "read_private", "read_public", "sequences", "write"];
+  permissions = [ "read", "read_private", "sequences", "write"];
```
Comment (jgadling) [inline src/backend/aspen/api/policy.polar:12]: the `read_public` permission made no sense, and even confused me writing tests. I was able to remove it once I wrote the tests, and then the tests and verified that worked properly.

File: src/backend/aspen/api/policy.polar [hunk 2]
```diff
@@ -58,6 +55,7 @@ resource PhyloRun {
   "read" if "viewer";
   # admin permissions
   "read" if "admin";
+  "write" if "admin";
```
Comment (jgadling) [inline src/backend/aspen/api/policy.polar:58]: Oops, admins should probably be able to edit phyloruns/trees 😅

File: src/backend/aspen/api/policy.polar [hunk 3]
```diff
@@ -76,14 +74,15 @@ resource PhyloTree {
...
   # member permissions
   "read" if "member";
   "write" if "member";
 }
 
 has_permission(ac: AuthContext, "read", sample: Sample) if
   has_permission(ac, "read_private", sample) or (
-    has_permission(ac, "read_public", sample) and
+    has_role(ac, "viewer", sample) and
```
Comment (jgadling) [inline src/backend/aspen/api/policy.polar:85]: Fix the policy for which samples you can see to not use the weird permission.

File: src/backend/aspen/api/utils/sample.py [hunk 4]
```diff
@@ -18,7 +18,7 @@ async def samples_by_identifiers(
 ) -> Query:
     # TODO, this query can be updated to use an "id in (select id from...)" clause when we get a chance to fix it.
     public_samples_query = (
-        (await az.authorized_query("read_public", Sample))
+        (await az.authorized_query("read", Sample))
```
Comment (jgadling) [inline src/backend/aspen/api/utils/sample.py:21]: The weird permission wasn't doing the right thing, let's use the better one here.

File: src/backend/aspen/api/views/tests/test_authz_policy.py [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+
+# All test coroutines will be treated as marked.
+pytestmark = pytest.mark.asyncio
+
+# Our user/group graph for these tests looks like this:
+# User0 -- member --> Group0
+# User1 -- member --> Group1 -- viewer --> Group2
+#                        +----- viewer --> Group3
+# User2 -- admin  --> Group2 -- member --> Group3 # We don't expect this use case right now, but let's make sure it works.
+# User3 -- admin  --> Group3 -- viewer --> Group2
```
Comment (lvreynoso) [inline src/backend/aspen/api/views/tests/test_authz_policy.py:29]: Very helpful!

File: src/backend/aspen/api/views/tests/test_authz_policy.py [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+        "North America", "USA", "California", "Santa Barbara County"
+    )
+
+    # Put 3 samples, trees, runs in each group.
+    objects_to_commit = []
+    trees: List[PhyloTree] = []
+    for i in range(len(groups)):
+        group = groups[i]
+        user = users[i]
+        # NOTE - this is horribly abusing python and adding arbitrary test data to our group models.
```
Comment (lvreynoso) [inline src/backend/aspen/api/views/tests/test_authz_policy.py:103]: I'm sure there's a more functional way to do this, but this approach is fine. (at least for now? It's clear enough for me.)

**Review by lvreynoso — APPROVED** [review]
Well done!

**Pull request merged**